lib/core/bufferFactory.cpp

```diff
@@ -100,8 +100,7 @@ struct Buffer* bufferFactory::new_buffer(const string& type_name, const string&
             num_prod = num_elements * (num_elements + 1) / 2;
         }
 
-        auto layout = visFrameView::calculate_buffer_layout(num_elements, num_prod, num_ev);
-        uint32_t frame_size = layout.first;
+        size_t frame_size = visFrameView::calculate_frame_size(num_elements, num_prod, num_ev);
```

Comment:
I'd be tempted to push the `config.get<>()` logic above into the `calculate_frame_size()` function.  i.e. keep implementation logic out of the factory.